docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+# Upgrade the templating toolset of Auth Server emails
+
+- Status: proposed
+- Deciders: Yogita Bhatia, Lauren Zugai, Jody Heavener, Vijay Budhram, [TBD]
+- Date: 2021-06-04
+
+Technical Story: [#8600](https://github.com/mozilla/fxa/issues/8600)
+
+## Context and Problem Statement
+
+[Mustache](https://www.npmjs.com/package/mustache), used in `fxa-auth-server` email templates, can accommodate very limited logic in its templates which has been a pain point for Firefox Accounts engineers and discussion around using a different templating system [began here](https://github.com/mozilla/fxa/issues/4627). While converting our emails to a more modern templating solution, there is an opportunity to evaluate what stack would be the most ideal for FxA emails beyond a proposed templating solution. This includes evaluating our CSS options, improving how we can preview emails in various states, and our localization tools, and the best approach for landing new templates in production.
+
+## Decisions and Considered Options
+
+### Templating
+
+- Option A - Continue to use Mustache
+- Option B - Use React server-side to generate static pages
+- Option C - Use EJS
+
+#### MJML
+
+We also considered whether or not to use the MJML responsive email framework.
+
+### CSS styles
+
+- Option A - Continue using ad-hoc inline styles
+- Option B - Use TailwindCSS
+- Option C - Use CSS-in-JS
+- Option D - Use a CSS pre-processor like SASS
+
+### Email previewing
+
+- Option A - Continue to use the `write-emails` command
+- Option B - Use Storybook
+- Option C - Use Mailtrap
+
+--
+
+Furthermore, there are a few **other decisions** worth noting that won’t necessarily have pros/cons lists:
+
+- How to handle the plaintext `.txt` files
```

Comment:
This wording is a little weird -- we're not handling TXT files per se, we are just generating plaintext versions of the emails.